src/main/java/com/marianbastiurea/lifeofbees/LifeOfBees.java

```diff
@@ -75,49 +91,48 @@ public void iterateOverTwoYears() {
 
         Calendar calendar = Calendar.getInstance();
         calendar.set(Calendar.YEAR, calendar.get(Calendar.YEAR) - 1); // Set the calendar to one year ago
-        calendar.set(Calendar.MONTH, Calendar.MARCH); // Start the year on March 1st
+        calendar.set(Calendar.MONTH, Calendar.APRIL); // Start the year on April 1st
         calendar.set(Calendar.DAY_OF_MONTH, 1);
 
-
         // Iterate over 2 years
         for (int year = 0; year < 1; year++) {// Use only one for debug purposes
-            while (calendar.get(Calendar.MONTH) != Calendar.APRIL) {
+            while (calendar.get(Calendar.MONTH) != Calendar.MAY) {
                 // Iterate until OCTOBER
                 Date currentDate = calendar.getTime();
                 System.out.println("Date: " + currentDate);
+                // Extract day of month and month from the calendar
+                int dayOfMonth = calendar.get(Calendar.DAY_OF_MONTH);
+                int monthValue = calendar.get(Calendar.MONTH);
+                HarvestingMonths month = HarvestingMonths.values()[monthValue];
 
 
-                for (Hive hive : apiary.getHives()) {
+                List<Hive> hives = apiary.getHives();
+                ArrayList<Hive> oldHives = new ArrayList<>(hives);
+                for (Hive hive : oldHives) {
                     Queen queen = new Queen();
-                    hive.addEggsBatches(queen.makeBatchOfEggs(queen.makeEggs(hive), currentDate));
+                    Honey honey=new Honey();
+                    hive.getHoney().setHoneyType("Rapeseed");
```

Comment:
Use a constructor with parameters to instantiate Honey instead of using setters.

Reply:
done